src/snarky.d.ts

```diff
@@ -51,489 +53,767 @@ declare interface ProvablePure<T> extends Provable<T> {
+   * 
+   * **Warning: This operation does NOT affect the circuit and can't be used to prove anything about the JSON string representation of the {@link Field}. Please use it only during debugging.
+   * 
+   * ```ts
+   * const someField = Field(42);
+   * console.log(Field.toJSON(someField));
+   * ```
+   * 
+   * @param value - The JSON string to coerce the {@link Field} from.
+   * 
+   * @return A string equivalent to the JSON representation of the given {@link Field}.
    */
-  static fromBits(x: (Bool | boolean)[]): Field;
-  /*
-  static toBits(x: Field | number | string | boolean): Bool[];
-  */
+  static toJSON(value: Field): string;
 
-  /*
-  static equal(
-    x: Field | number | string | boolean,
-    y: Field | number | string | boolean
-  ): Bool;
-  */
+  /**
+   * Deserialize a JSON string containing a `field-like` value into a {@link Field} element.
+   * 
+   * **Warning: This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the {@link Field}.
+   * 
+   * @param value - the `field-like` value to coerce the {@link Field} from.
+   * 
+   * @return A {@link Field} coerced from the given JSON string.
+   */
+  static fromJSON(value: string): Field;
 
   /**
-   * Serialize a {@link Field} to a JSON string.
-   * This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the Field.
+   * This function is the implementation of {@link Provable.check} in {@link Field} type.
+   * 
+   * As any {@link Provable} type can be a {@link Field}, this function does not create any assertions on the chain, so it basically does nothing :)
+   * 
+   * @param value - the {@link Field} element to check.
    */
-  static toJSON(x: Field): string;
+  static check(value: Field): void;
 
   /**
-   * Deserialize a JSON structure into a {@link Field}.
-   * This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the Field.
+   * **Warning: This function is mainly for internal use. Normally it is not intended to be used by a zkApp developer.
+   * 
+   * This function is the implementation of {@link Provable.toInput} in {@link Field} type.
+   * This function acts identical to {@link Field.toFields} for {@link Field} type.
+   * 
+   * @param value - The {@link Field} element to get the `input` array.
+   * 
+   * @return A {@link Field} array of length 1 created from this {@link Field}.
+   * 
    */
-  static fromJSON(x: string): Field;
+  static toInput(value: Field): { fields: Field[] };
 
-  static check(x: Field): void;
+  /**
+   * Create an array of digits equal to the [little-endian](https://betterexplained.com/articles/understanding-big-and-little-endian-byte-order/) byte order of the given {@link Field} element.
```

Comment:
Do you mind replacing the link with a website that could be more "permanent"? I just worry about the chance of referencing a website and then having the link 404.  Wikipedia is probably good enough to link: https://en.wikipedia.org/wiki/Endianness

What do you think?